cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -1221,6 +1317,16 @@ const FunctionDoc atan2_doc{"Compute the inverse tangent of y/x",
                             ("The return value is in the range [-pi, pi]."),
                             {"y", "x"}};
 
+const FunctionDoc atanh_doc{"Compute the inverse hyperbolic tangent",
+                            ("NaN is returned for invalid input values;\n"
+                             "to raise an error instead, see \"atanh_checked\"."),
+                            {"x"}};
```

Comment:
```suggestion
const FunctionDoc atanh_doc{"Compute the inverse hyperbolic tangent",
                            ("NaN is returned for input values outside of the (-1.0, 1.0) range;\n"
                             "to raise an error instead, see \"atanh_checked\"."),
                            {"x"}};
```

Reply:
Updated the suggestion to mention -Inf/Inf

Reply:
Chose slightly different wording, but same spriti.